### PR TITLE
Feat/swiglu

### PR DESCRIPTION
**Pull request by ashdtu**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

#1504 

### Changes

Implement [SwiGLU activation](https://paperswithcode.com/method/swiglu) module to be used in LLM models

### CODE REVIEW

**Review by antimora — CHANGES_REQUESTED** [review]
It looks but tests are missing for some reason.

File: crates/burn-core/src/nn/swiglu.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+use crate::config::Config;
+use crate::module::Module;
+use crate::tensor::activation::silu;
+use crate::tensor::{backend::Backend, Tensor};
+
+use super::{Initializer, Linear, LinearConfig};
+
+/// Configuration to create a [SwiGLU](SwiGLU) activation layer.
+#[derive(Config, Debug)]
+pub struct SwiGLUConfig {
```
Comment (louisfd) [inline crates/burn-core/src/nn/swiglu.rs:12]: Should be camel case, see #1316

**Review by louisfd — APPROVED** [review]
LGTM. One small comment but approved in advance

File: crates/burn-core/src/nn/swiglu.rs (hunk 1 above)
Comment (ashdtu) [reply crates/burn-core/src/nn/swiglu.rs]: Done! thanks

**Comment by ashdtu** [comment]
@antimora Added tests. Let me know if it looks good to merge.

**Review by antimora — APPROVED** [review]
LGTM

**Pull request merged**